### PR TITLE
Linux driver maintenance

### PR DESCRIPTION
**Pull request by ajs-sun**
This commit does the following:
- clean unnecessary code like unused files, redundant struct member, etc
- re-factor some functions
- fix code style
- clean up trace for debug
- fix spell mistakes
- change location of some macros

Signed-off-by: Aijun Sun <aijun.sun@linaro.org>

### CODE REVIEW

**Comment by jbech-linaro** [comment]
I've only done a quick review here, but since the changes are coming from the Aijun's branch where I've done a quite thorough review, I'm happy with this as it it here.

`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: Kconfig [hunk 1]
```diff
@@ -11,14 +11,15 @@
...
-# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
-#
+# with this program.  If not, see <http://www.gnu.org/licenses/>
 
 # Trursted Execution Environment Configuration
-config TEE_SUPPORT
-	bool "Trusted Execution Environment Support"
-	default y
+config  OPTEE
+	tristate "OPTEE driver support"
```
Comment (jforissier) [inline Kconfig]: OP-TEE

File: Kconfig [hunk 2]
```diff
@@ -11,14 +11,15 @@
...
+        depends on ARM64 || ARM
+        select DMA_SHARED_BUFFER
+	default n
 	---help---
-	  This implements the Trusted Execution Environment (TEE) Client
-	  API Specification from GlobalPlatform Device Technology.
+	  This option implements the kernel driver for OP-TEE. If you want to
```
Comment (jforissier) [inline Kconfig]: Add link to http://github.com/OP-TEE and give the meaning of TEE.

File: armtz/tee_mem.c [hunk 3]
```diff
@@ -308,15 +310,17 @@ void *tee_shm_pool_p2v(struct device *dev, struct shm_pool *pool,
 	if (!is_valid_paddr(pool, paddr)) {
 		mutex_unlock(&pool->lock);
 		dev_err(dev,
-			"tee_shm_pool_p2v() paddr=0x%p not in the shm pool\n",
+			"tee_shm_pool_p2v() paddr(0x%p) is out of shm area\n",
```
Comment (jforissier) [inline armtz/tee_mem.c]: dev_err() messages should not be formatted like debug messages, see https://www.kernel.org/doc/Documentation/CodingStyle chapter 13 "Printing kernel messages". This one should probably be a dev_dbg().

File: armtz/tee_tz_drv.c [hunk 4]
```diff
@@ -1205,78 +1111,60 @@ static int tz_tee_remove(struct platform_device *pdev)
...
-		},
 };
 
 static int __init tee_tz_init(void)
 {
-	int rc;
+	int ret;
 
-	pr_info("TEE armv7 Driver initialization\n");
+	pr_info("OP-TEE ARMv7 driver initialization\n");
```
Comment (jforissier) [inline armtz/tee_tz_drv.c]: Works on ARMv8 too

File: core/tee_core_priv.h [hunk 5]
```diff
@@ -26,7 +25,7 @@ struct tee_shm *tee_context_create_tmpref_buffer(struct tee_context *ctx,
 						 const void *buffer, int type);
 struct tee_shm *tee_context_alloc_shm_tmp(struct tee_context *ctx, size_t size,
 					  const void *data, int type);
-int tee_context_copy_from_client(const struct tee_context *ctx, void *dest,
+int tee_context_copy(bool dest, struct tee_context *ctx, void *to,
```
Comment (jforissier) [inline core/tee_core_priv.h]: s/dest/from_user/ for clarity

**Comment by ajs-sun** [comment]
The new patch has been sent to linux mailist. Once it's accepted, the driver will not be maintained any more. So it's no need to merge such a big patch.

**Pull request closed without merging**